### PR TITLE
Added details for connecting Knative Services to existing K8 deployments

### PR DESCRIPTION
**Pull request by abrennan89**

### CODE REVIEW

File: modules/knative-services-connectivity.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// * serverless/getting-started-knative-services.adoc
+
+[id="knative-services-connectivity_{context}"]
+= Connecting Knative Services to existing K8 deployments
+
```
Comment (wtam2018) [inline modules/knative-services-connectivity.adoc]: Replace "K8" with "K8s" or "Kubernetes"

File: modules/knative-services-connectivity.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Knative Services can call a K8 deployment in any namespace, provided that there are no existing additional network barriers.
+
+A K8 deployment can call a Knative Service if:
+
+* The K8 deployment is in the same namespace as another Knative Service.
```
Comment (wtam2018) [inline modules/knative-services-connectivity.adoc]: Is "as target Knative Service" clearer?

**Review by wtam2018 — APPROVED** [review]

File: modules/knative-services-connectivity.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+[id="knative-services-connectivity_{context}"]
+= Connecting Knative Services to existing Kubernetes deployments
+
+Knative Services can call a Kubernetes deployment in any namespace, provided that there are no existing additional network barriers.
+
+A Kubernetes deployment can call a Knative Service if:
+
+* The Kubernetes deployment is in the same namespace as the target Knative Service.
+* The Kubernetes deployment is in a namespace that has manually been added to the ServiceMeshMemberRoll.
+* The Kubernetes deployment uses the target Knative Service's public URL.
```
Comment (markusthoemmes) [inline modules/knative-services-connectivity.adoc]:
```suggestion
* The Kubernetes deployment uses the target Knative Service's public URL (implying that the Knative Service is not configured to be cluster-local only).
```

File: modules/knative-services-connectivity.adoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+[id="knative-services-connectivity_{context}"]
+= Connecting Knative Services to existing Kubernetes deployments
+
+Knative Services can call a Kubernetes deployment in any namespace, provided that there are no existing additional network barriers.
+
+A Kubernetes deployment can call a Knative Service if:
+
+* The Kubernetes deployment is in the same namespace as the target Knative Service.
+* The Kubernetes deployment is in a namespace that has manually been added to the ServiceMeshMemberRoll.
```
Comment (markusthoemmes) [inline modules/knative-services-connectivity.adoc]:
```suggestion
* The Kubernetes deployment is in a namespace that has manually been added to the ServiceMeshMemberRoll in `knative-serving-ingress`.
```

**Review by markusthoemmes — COMMENTED** [review]
Two minor additions.

File: modules/knative-services-connectivity.adoc (hunk 4 above)
Comment (abrennan89) [reply modules/knative-services-connectivity.adoc]: Do we need to say anything else about `knative-serving-ingress`? I'm not sure we mention this elsewhere in the docs. What is this, why is it required and how does it come to exist?

File: modules/knative-services-connectivity.adoc (hunk 3 above)
Comment (abrennan89) [reply modules/knative-services-connectivity.adoc]: Added a note, lmk what you think.

**Comment by abrennan89** [comment]
Updated the docs and added some comment responses :slightly_smiling_face: Can you please take another look and confirm @markusthoemmes ?

File: modules/knative-services-connectivity.adoc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+A Kubernetes deployment can call a Knative Service if:
+
+* The Kubernetes deployment is in the same namespace as the target Knative Service.
+* The Kubernetes deployment is in a namespace that has manually been added to the ServiceMeshMemberRoll in `knative-serving-ingress`.
+* The Kubernetes deployment uses the target Knative Service's public URL.
+
+[NOTE]
+====
+Knative Services can be accessed by using a public URL by default. The target Knative Service must not be configured as cluster-local only so that the Kubernetes deployment can access the Service using the public URL.
```
Comment (markusthoemmes) [inline modules/knative-services-connectivity.adoc]: This note only applies to the last bullet above. It only applies if the Kubernetes deployment is not in the same ServiceMeshMemberRoll as the Knative Service (which is provided by the first two bullets above).

If a K8s deployment is in the same namespace as the ksvc or in the same smmr, a cluster-local service works.

Comment (abrennan89) [reply modules/knative-services-connectivity.adoc]: Fixed formatting so this is part of the bullet

**Comment by maschmid** [comment]
QE approved

File: modules/knative-services-connectivity.adoc [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+[id="knative-services-connectivity_{context}"]
+= Connecting Knative Services to existing Kubernetes deployments
+
+Knative Services can call a Kubernetes deployment in any namespace, provided that there are no existing additional network barriers.
+
+A Kubernetes deployment can call a Knative Service if:
+
+* The Kubernetes deployment is in the same namespace as the target Knative Service.
+* The Kubernetes deployment is in a namespace that has manually been added to the ServiceMeshMemberRoll in `knative-serving-ingress`.
```
Comment (ahardin-rh) [inline modules/knative-services-connectivity.adoc]: s/has manually been added/was manually added

File: modules/knative-services-connectivity.adoc [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+* The Kubernetes deployment uses the target Knative Service's public URL.
++
+[NOTE]
+====
+Knative Services can be accessed by using a public URL by default. The target Knative Service must not be configured as cluster-local only so that the Kubernetes deployment can access the Service using the public URL.
```
Comment (ahardin-rh) [inline modules/knative-services-connectivity.adoc]: By default, you can access Knative Services using a public URL.

Should "cluster-local only" be marked up? I assume it's a setting or a menu option.

**Comment by bergerhoffer** [comment]
Cherrypicking to 4.3 and master, since this was checked out against 4.2. But as discussed in the future, if it applies to the latest (4.3) and backwards, best to start against master.

**Pull request merged**